Library.Infrastructure.Mongo/Repositories/BookCheckoutMongoRepository.cs

```diff
@@ -0,0 +1,95 @@
+﻿using Library.Domain.Interfaces;
+using Library.Domain.Models;
+using MongoDB.Driver;
+
+namespace Library.Infrastructure.Mongo.Repositories;
+
+/// <summary>
+/// Репозиторий с CRUD-операциями для объектов BookCheckout для работы с MongoDb.
+/// </summary>
+public class BookCheckoutMongoRepository: IRepositoryAsync<BookCheckout, int>
+{
+    private readonly IMongoCollection<BookCheckout> _checks;
+    private int _maxId;
+
+    /// <summary>
+    /// Конструктор для репозитория. Получает коллекцию выдач из контекста MongoDB.
+    /// и определяет текущий максимальный Id, для добавления новых объктов.
+    /// </summary>
+    /// <param name="context">Контекст MongoDB</param>
+    public BookCheckoutMongoRepository(MongoDbContext context)
+    {
+        _checks = context.GetCollection<BookCheckout>("checkouts");
+        var lastTask = _checks.Find(_ => true)         
+                      .SortByDescending(x => x.Id)  
+                      .FirstOrDefaultAsync();    
```

Comment:
Выровнять код. Привычнее смотреть на код если он отделен от верхнего уровня одной табуляцией

```suggestion
        var lastTask = _checks.Find(_ => true)
            .SortByDescending(x => x.Id)
            .FirstOrDefaultAsync();
```